pylint/checkers/stdlib.py

```diff
@@ -599,6 +620,35 @@ def _check_lru_cache_decorators(self, decorators: nodes.Decorators) -> None:
                 confidence=interfaces.INFERENCE,
             )
 
+    def _check_dispatch_decorators(self, node: nodes.FunctionDef) -> None:
+        decorators_names_set: set[str] = set()
+        decorators_name_node_map: dict[str, nodes.NodeNG] = {}
+
+        decorators = node.decorators.nodes if node.decorators else []
+        for decorator in decorators:
```

Comment:
We could use `inference` here as users might overwrite `classmethod`. But I think in this case we probably shouldn't. Overwriting `exit` or `print` I can sort of understand but if you're overwriting `classmethod` you probably also don't use `pylint` to make sure you have sane code :sweat_smile: